Payroll rubric config

lomamech opened the pull request:
- Create Tables config_rubric and config_rubric_item
- Implement API for Configuration of Payroll Rubrics usable in Period of
  paie
- Implementation of the creation of the configuration of the Payroll rubrics and assignment of the items to a
Configuration
- Implementation API for read and set payroll rubric's configuration
- Improve Rubrics Configuration Display
- Implementation Integration Test for Payroll Rubric Configuration
- Add property membership fee (Cotisation)
- Update and Improve form for Rubric Management
- Update API for rubric management with the property is_membership_fee
- (test) update integration and End to End test for Rubric Management
- refactor Payroll Rubric Configuration, add property Membership Fee in
  configuration display
- Complete Integration test for Rubric Configuration
- Fix configuration Rubrics Modal
- Implement test E2E for payroll Rubric Configuration
- Update and Improve some file

reference #2292

lomamech committed:
refactor(Payroll Configuration)
- Create Tables config_rubric and config_rubric_item

lomamech committed:
feature(Payroll Period Configuration)
- Implementation of the creation of the configuration of the Payroll rubrics and assignment of the items to a
Configuration
- Implementation API for read and set payroll rubric's configuration

lomamech committed:
Typescript

lomamech committed:
refactor(PayrollConfiguration)
- Sanitaze Code
- Improve Rubrics Configuration Display

lomamech committed:
test(Payroll Rubric)
- Implementation Integration Test for Payroll Rubric Configuration

lomamech committed:
refactor(Payroll Rubric)
- Add property membership fee (Cotisation)
- Update and Improve form for Rubric Management
- Update API for rubric management with the property is_membership_fee
- (test) update integration and End to End test for Rubric Management

lomamech committed:
refactor(Payroll Rubric Configuration)
- refactor Payroll Rubric Configuration, add property Membership Fee in
  configuration display

lomamech committed:
test(Payroll Rubric Configuration)
- Complete Integration test for Rubric Configuration
- Fix configuration Rubrics Modal

lomamech committed:
test(Payroll Rubric)
- Implement test E2E for payroll Rubric Configuration
- Update and Improve some file

jniles commented on client/src/modules/payroll/rubric_configuration/configuration.js:
```diff
@@ -0,0 +1,102 @@
+  }
+
+  // update an existing Rubric Configuration
+  function editConfig(title) {
```
To reduce the amount of JS in the controller, you could try using `ui-sref`s in the templates for these.  It should be almost the same.

jniles commented on client/src/modules/payroll/rubric_configuration/configuration.service.js:
```diff
@@ -0,0 +1,32 @@
+
+  // loads the configuration's rubrics
+  function getRubrics(id) {
+    return $http.get(`/rubric_config/${id}/setting`)
```
Nice use of template strings 👍

jniles commented on client/src/modules/payroll/rubric_configuration/configuration.service.js:
```diff
@@ -0,0 +1,32 @@
+ * @description
+ * Encapsulates common requests to the /rubric_config/ URL.
+ */
+function ConfigurationService(Api, Modal, $http, util) {
```
Do you need `Modal` here?

jniles commented on client/src/modules/payroll/rubric_configuration/modals/config.modal.js:
```diff
@@ -0,0 +1,208 @@
+  Rubrics.read()
+    .then(function (rubrics) {
+      vm.rubrics = rubrics;
+      vm.socialCares = rubrics.filter(function (item) {
```
Just FYI this could be written:
```js
vm.socialCares = rubrics.filter(item => item.is_social_care);
```

jniles commented on client/src/modules/payroll/rubric_configuration/modals/config.modal.js:
```diff
@@ -0,0 +1,208 @@
+  '$state', 'ConfigurationService', 'ModalService', 'NotifyService', 'appcache', 'RubricService',
+];
+
+function RubricConfigModalController($state, Configs, ModalService, Notify, AppCache, Rubrics) {
```
It doesn't look like ModalService is used here.

jniles commented on client/src/modules/payroll/rubric_configuration/templates/action.tmpl.html:
```diff
@@ -0,0 +1,27 @@
+    <span class="caret"></span>
+  </a>
+
+  <ul data-action="{{rowRenderIndex}}"  class="dropdown-menu-right" bh-dropdown-menu-auto-dropup uib-dropdown-menu>
```
👍

jniles commented on client/src/modules/payroll/rubrics/modals/rubric.modal.js:
```diff
@@ -50,13 +48,13 @@ function RubricModalController($state, Rubrics, ModalService, Notify, AppCache)
     }
 
-    if(vm.rubric.is_discount === "D"){
+    if(vm.rubric.is_discount){
```
Nice 👍

jniles reviewed (approved):
LGTM!  I made a couple of comments, but none of them blocking.

bors r+

lomamech committed:
refactor(Payroll Rubric Configuration)
- Use ui-sref instead function with $state.go
- Re write functions filter

lomamech committed:
Resolve conflict with Branch Master

lomamech committed:
Resolve conflict with Branch Master

lomamech committed:
refactor(Payroll Rubric)
- Use formating of data instead using of templates for debtor and expense
  account

lomamech committed:
Rename file in end to end repertory

lomamech committed:
refactor(Payroll Rubric)
- Fix problem with Common funtion name
